### PR TITLE
[JUJU-3568] Add podspec deprecation warning to CLI

### PR DESCRIPTION
**Pull request by barrettj12**
#14162 added this on the server-side, but now we make it more visible, by printing a warning to stdout when you `juju deploy` a podspec charm.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing
- ~[ ] [Integration tests](https://github.com/juju/juju/tree/develop/tests), with comments saying what you're testing~
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

Bootstrap a k8s controller.

### Deploy from Charmhub
```console
$ juju deploy redis-k8s
Located charm "redis-k8s" in charm-hub, revision 7
WARNING deploying podspec charm "redis-k8s": podspec charms are deprecated. Support for them will be removed soon.
Deploying "redis-k8s" from charm-hub charm "redis-k8s", revision 7 in channel stable on ubuntu@20.04/stable
```

### Deploy local charm
```console
$ juju download redis-k8s
$ juju deploy ./redis-k8s_r7.charm --resource redis-image=redis
Located local charm "redis-k8s", revision 0
WARNING deploying podspec charm "redis-k8s": podspec charms are deprecated. Support for them will be removed soon.
Deploying "redis-k8s" from local charm "redis-k8s", revision 0 on ubuntu@20.04/stable
```

### Deploy bundle containing podspec charm

Create a file `bundle.yaml` with the following contents:
```yaml
applications:
  redis:
    charm: redis-k8s
```
```console
$ juju deploy ./bundle.yaml
Located charm "redis-k8s" in charm-hub, channel stable
Executing changes:
- upload charm redis-k8s from charm-hub with architecture=amd64
- deploy application redis from charm-hub using redis-k8s
WARNING deploying podspec charm "redis-k8s": podspec charms are deprecated. Support for them will be removed soon.
  added resource redis-image
Deploy of bundle completed.
```

### CODE REVIEW

**Review by hpidcock — APPROVED** [review]
LGTM - will wait for review from @tmihoc on the wording. @hmlanigan can we replicate this with server side deploy?

**Comment by tmihoc** [comment]
Regarding the wording:

>  <charm> is a deprecated podspec charm. Support for podspec charms will be removed soon.

To me, "is a deprecated podspec charm" is awkward because it can have two readings -- (1) a podspec charm that is deprecated as opposed to other podspec charms that may not be and (2) one of the set of podspec charms, which is a deprecated set -- and my brain doesn't want to let go of the former, even as I know that what we mean is the latter.

I'm also not happy with the vagueness in "soon". 

How about:

> <charm> is a podspec charm. Podspec charms were deprecated in Juju `v.3.0`. Support for them will be dropped in Juju ...

I don't love it but it avoids the issues I mentioned. Though I understand if we can't be precise about when we'll drop support. Also, if nobody else is bothered by those issues in the first place.

**Pull request merged**
